portal/account/register.php

```diff
@@ -0,0 +1,556 @@
+                flg = 1;
+            }
+            $(curInputs[i]).closest(".form-group").addClass("has-error");
+        }
+    }
+
+    if (!isValid) {
+        return false;
+    }
+
+    return true;
+}
+
+function callServer(action, value, value2, last, first) {
+    var data = {
+        'action' : action,
+        'value' : value,
+        'dob' : value2,
+        'last' : last,
+        'first' : first
+    }
+    if (action == 'do_signup') {
+        data = {
+            'action': action,
+            'pid': value
+        };
+    }
+    if (action == 'notify_admin') {
+        data = {
+            'action': action,
+            'pid': value,
+            'provider': value2
+        };
+    }
+    if (action == 'cleanup') {
+        data = {
+            'action': action
+        };
+    }
+    // The magic that is jquery ajax.l
+    $.ajax({
+        type : 'GET',
+        url : 'account.lib.php',
+        data : data
+    }).done(function (rtn) {
+        if (action == "cleanup") {
+            window.location.href = "./../index.php" // Goto landing page.
+        }
+        if (action == "set_lang") {
+            window.location.href = window.location.href;
+        }
+        if (action == "get_newpid") {
+            if (parseInt(rtn) > 0) { // set pid for restfull save
+                newPid = rtn;
+                $("#profileFrame").contents().find('input#pubpid').val(newPid);
+                $("#profileFrame").contents().find('input#pid').val(newPid);
+            } else { // After error alert app exit to landing page.
+                eModal.alert(rtn); // Existing user error. Error message is translated in account.lib.php. :)
+            }
+        }
+        if (action == 'do_signup') {
+            if (rtn == "") {
+                var message = "<?php echo xlt('Unable to either create credentials or send email.'); ?>";
+                alert(message);
+                return false;
+            }
+
+            // For production. Here we're finished so do signup closing alert and then cleanup.
+            callServer('notify_admin', newPid, provider); // pnote notify to selected provider
+            // I left alert below for ease of testing.
+            //alert(rtn); // sync alert.. rtn holds username and password for testing. @TODO disable for production.
+
+            var message = "<?php echo xlt(" Your new credentials have been sent.Check your email inbox and also possibly your spam folder." .
+                                    " Once you log into your patient portal feel free to make an appointment or send us a secure message." . " We look forward to seeing you soon."); ?>"
+
+                eModal.alert(message); // This is an async call. The modal close event exits us to portal landing page after cleanup.
+
+        }
+    }).fail(function (err) { // @TODO Where nobody wants to go. Maybe something better here besides ooops!
+        var message = "<?php echo xlt('Something went wrong.' + " "+ err) ?>";
```

Comment:
Don't include the err stuff in the translation constant. Otherwise will be impossible to translate.
use xls()